pysr/regressor_sequence.py

```diff
@@ -0,0 +1,189 @@
+            If `X` is a pandas dataframe, the column name will be used
+            instead of `variable_names`. Cannot contain spaces or special
+            characters. Avoid variable names which are also
+            function names in `sympy`, such as "N".
+            The number of variable names must be equal to (n_features,).
+        X_units : list[str]
+            A list of units for each variable in `X`. Each unit should be
+            a string representing a Julia expression. See DynamicQuantities.jl
+            https://symbolicml.org/DynamicQuantities.jl/dev/units/ for more
+            information.
+            Length should be equal to n_features.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+
+        _check_assertions(
+            X,
+            y,
+            Xresampled,
+            self.recursive_history_length,
+            weights,
+            variable_names,
+            X_units,
+            y_units,
+        )
+
+        historical_X = X[self.recursive_history_length :]
+        current_X = np.lib.stride_tricks.sliding_window_view(
+            X[:-1].flatten(), self.recursive_history_length * X.shape[1]
+        )[:: historical_X.shape[1], :]
+        y_units = X_units
+        if isinstance(weights, np.ndarray):
+            weights = weights[self.recursive_history_length :]
+        variable_names = self._variable_names(y, variable_names)
+
+        super().fit(
+            X=current_X,
+            y=historical_X,
+            weights=weights,
+            variable_names=variable_names,
+            X_units=X_units,
+            y_units=y_units,
+            complexity_of_variables=complexity_of_variables,
+        )
+
+        return self
+
+    def predict(self, X, index=None):
+        """
+        Predict y from input X using the equation chosen by `model_selection`.
+
+        You may see what equation is used by printing this object. X should
+        have the same columns as the training data.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Data of shape `(n_times, n_features)`.
+        index : int | list[int]
+            If you want to compute the output of an expression using a
+            particular row of `self.equations_`, you may specify the index here.
+            For multiple output equations, you must pass a list of indices
+            in the same order.
+
+        Returns
+        -------
+        x_predicted : ndarray of shape (n_samples, n_features)
+            Values predicted by substituting `X` into the fitted sequence symbolic
+            regression model.
+
+        Raises
+        ------
+        ValueError
+            Raises if the `best_equation` cannot be evaluated.
+        """
+        _check_assertions(X, self.recursive_history_length)
+        current_X = np.lib.stride_tricks.sliding_window_view(
```

Comment:
We are _predicting_ the `current_X` based on historical data so this should be `historical_X`